common/src/main/java/com/microsoft/identity/common/internal/cache/IAccountCredentialCache.java

```diff
@@ -17,11 +20,28 @@
 
     List<Account> getAccounts();
 
+    List<Account> getAccounts(
+            @Nullable final String uniqueId,
+            final String environment,
+            @Nullable final String realm
+    );
+
     List<Credential> getCredentials();
 
-    void clearAccounts();
+    List<Credential> getCredentials(
+            @Nullable final String uniqueId,
+            final String environment,
+            final CredentialType credentialType,
+            final String clientId,
+            @Nullable final String realm,
+            @Nullable final String target
+    );
+
+    boolean removeAccount(final String uniqueId, final String environment);
+
+    boolean removeCredential(final Credential credentialToClear);
 
-    void clearCredentials();
+    int removeAll(final String uniqueId, final String environment);
 
     void clearAll();
```

Comment:
@oldalton Can't get rid of this function yet, as I need it for clearing the cache during unit testing. From our conversation earlier, I doubt we should ship with this function in place.